### ISSUE TITLE
Full-Page Translation

### ISSUE THREAD

**Issue by alexpong0630** [Royal-lobster/Syncia#87]
I've added a full-page translation feature for Syncia on my private repo.

I'm curious if you're interested in this new feature, since Syncia can already perform selected text translations through custom prompts.

P.S. 
- I'm not familiar with JS, so if integration is needed, I might require you to rewrite the source code.
- Not guaranteed the translation will show under the original text
- Not guaranteed all content will be translated

Some screenshots:

**Translation Result:**
![image](https://github.com/user-attachments/assets/177f0457-683d-41e2-8a58-0ed30343ea5f)

**Page Translation Flag**
![image](https://github.com/user-attachments/assets/269af54e-dd51-453c-a6f6-fdc4c4bb2545)

**Translation Settings**
![image](https://github.com/user-attachments/assets/9456eaf0-4601-4c9b-b63a-1a5ad43e4e6d)

**Supported Languages (GPT3.5 supported languages(copy from the internet))**
![image](https://github.com/user-attachments/assets/eaff2dda-c44c-414c-a10b-098460f7b316)

**Comment by Royal-lobster** [comment]
Interesting ! tho we need to think about this, how fast is the translation ? is it better than the google translate ? cuz i think its much more faster than llm translation

**Comment by alexpong0630** [comment]
My original intention was to find a chrome extension that can translate while retaining the original text. 

The chrome web store does have such extensions, but they are not open source, so I cannot ensure that they use the data records for other purposes when accessing my content.

There are some examples:

[deep translate](https://chromewebstore.google.com/detail/deeptranslate-full-page-b/noofnfniimcnghnkonpdlhlaeaocidbi?hl=en-US&utm_source=ext_sidebar)

[immersive translate](https://immersivetranslate.com/)


**how fast is the translation ?** 
Not faster than google translate because it use your LLM to translate currently.

**is it better than the google translate ?**
Not sure. It depends on your LLM

**Comment by Royal-lobster** [comment]
Makes sense. I am thinking to create a plugin system for syncia, something similar to raycast (raycast.com/) so we can expose the webpage content, and some actions to the plugin code to manipulate syncia behaviour. perhaps this can be a plugin
